lib/developer_portal/lib/cms/toolbar.rb

```diff
@@ -39,6 +40,17 @@ def draft?
 
   private
 
+  def extract_cms_token(encrypted_token)
+    return encrypted_token if encrypted_token.blank?
+
+    encryptor = ThreeScale::SSO::Encryptor.new(site_account.settings.sso_key)
+    encryptor.extract!(encrypted_token).first
+  rescue StandardError
+    # When the token is invalid, I don't think we have to bother the client and BugSnag with an exception
+    # Better return an empty token which means "Disable CMS edit mode (hide toolbar)"
+    ''
```

Comment:
I wonder if we can show a warning that edit mode was disabled due to invalid or expired token.

Reply:
I added a flash error: https://github.com/3scale/porta/pull/3932/commits/d5420291cbbf9957bc53620aa175d83d29c83ba3